renku/core/dataset/providers/s3.py

```diff
@@ -59,6 +65,43 @@ def supports_create() -> bool:
         """Whether this provider supports creating a dataset."""
         return True
 
+    @staticmethod
+    def supports_add() -> bool:
+        """Whether this provider supports adding data to datasets."""
+        return True
+
+    @staticmethod
+    def add(client: "LocalClient", uri: str, destination: Path, **kwargs) -> List["DatasetAddMetadata"]:
+        """Add files from a URI to a dataset."""
+        if re.match(r"\*|\?", uri):
+            raise errors.ParameterError("Wildcards like '*' or '?' are not supported in the uri for S3 datasets.")
+        provider = S3Provider(uri=uri)
+        credentials = S3Credentials(provider=provider)
+        prompt_for_credentials(credentials)
+
+        storage = get_storage(provider=provider, credentials=credentials)
+        dataset = kwargs.get("dataset")
+        if dataset and dataset.storage and not is_uri_subfolder(dataset.storage, uri):
+            raise errors.ParameterError(
+                f"S3 uri {uri} should be located within or at the storage uri {dataset.storage}. "
+            )
+        if not storage.exists(uri):
+            raise errors.ParameterError(f"S3 bucket '{uri}' doesn't exists.")
+        storage.mount(uri, mount_location=destination)
```

Comment:
We shouldn't mount or download files at this stage. This operation is only about adding files to the dataset and not copying them. I guess this is required for hash calculation if rclone cannot hash a file; in this case, it's better not to have a hash and then add it later when we pull/copy/stage the data.

Reply:
I will remove this. I did not understand the issue.